core/src/com/unciv/logic/map/MapUnit.kt

```diff
@@ -283,8 +283,10 @@ class MapUnit {
 
         //todo: consider parameterizing [terrainFilter] in some of the following:
         canEnterIceTiles = hasUnique(UniqueType.CanEnterIceTiles)
-        cannotEnterOceanTiles = hasUnique(UniqueType.CannotEnterOcean)
-        cannotEnterOceanTilesUntilAstronomy = hasUnique(UniqueType.CannotEnterOceanUntilAstronomy)
+        cannotEnterOceanTiles = hasUnique(UniqueType.CannotEnterOcean, StateForConditionals(civInfo=civInfo, unit=this))
```

Comment:
Does this update when we gain the required tech?

Reply:
Yes, but there are bigger problems. It doesn't currently work at all, as the 'cannot enter ocean tiles' does not apply before researching astronomy, so you can always enter ocean tiles. After researching Astronomy you have researched Astronomy, so you can enter ocean tiles.
I'll probably need to rework most of this PR for this to work correctly, until then I'll mark this as draft.

Reply:
I've returned to this PR and found that what I thought was happening, did in fact not happen, but I just made a typo.

This also doesn't update when we gain the required tech, but that behaviour is not changed from before this PR: if you researched astronomy mid-turn your ships also wouldn't be able to enter ocean tiles immediately. If need be I can implement this, but it will cost performance, as the only way to do this correctly would be to update the uniques of all units every time a tech is discovered